### ISSUE TITLE
Accounting for directionality of dataset data-BioGrid-Yeast

### ISSUE THREAD

**Issue by T-Wisse** [T-Wisse/MEP_Thomas#8]
I have been messing around with your scripts, and for interaction data it uses the excel file data-BioGrid-Yeast, which has interaction data for a large number of genes. There are a number of duplicates in the file but that is not a problem. While testing with BEM2 I observed that the list of BEM2 as query is not identical to the list of BEM2 as a target. This makes sense as the dataset is large and you don't want duplicate entries. However, in the code it only searches for the query gene (BEM2) in the query column, which causes a number of interactions to be missed.  For now I took the easy way out and simply duplicated the dataset, switching query and target columns, which results in quite a different figure. I am not sure if you were actually using this dataset but its good to know.
![Figure 2020-12-16 093057](https://user-images.githubusercontent.com/70693768/102326112-445b5000-3f84-11eb-8b08-3c798ef04b08.png) 
![Figure 2020-12-15 083822](https://user-images.githubusercontent.com/70693768/102326146-51783f00-3f84-11eb-8669-072a71918af3.png)

**Comment by leilaicruz** [comment]
Hi Thomas, the point here is to know what are the concepts behind `query` and `target`, which refer to the way SGA works. I would read first again that part of the "Genetic Networks" by Constanzo. In general , I take the query column as the gene of interest and then look to the unique values in the target column as their interactors. The query is the initial mutation and the target are the double deletion genotypes found after sporulation, and are the ones they could measure the fitness of. Another point is that  genetic interactions are not bidirectional, that is  that if A is positive interactor of B in certain background does not mean that  B is a positive interactor of A, for example the case of bem1 and bem3. bem3 is a positive interactor of bem1 however bem1 is not a positive interactor of bem3 within the same definition(dbem1dbem3 growth rate < dbem3 growth rate). 
Hence that is why I took as a convention to always look the interactors in the target column that correspond to certain gene in the query column. 
does that make sense to you?

**Comment by T-Wisse** [comment]
The part about SGA makes sense yes. However the way I understood it the genetic interaction should be bidirectional. That dbem1dbem3 growth rate < dbem3 growth rate does not mean they do not positively interact. It should be dbem1dbem3 growth rate <= Expected dbem1dbem3 growth rate. Basically that while introducing the bem1 delete to the bem3 delete does lower the fitness, the interaction is positive because it does not lower as much as expected (as seen in wild type).  If I made an error in my thinking there you are right to do it that way of course.

**Comment by leilaicruz** [comment]
Yes , what you said is true according the mathematical definition.  But
practically,  for the case of bem1 and bem3 , the deletion of bem1 do not
rescue the bem3deleted phenotype. While the bem3 deletion does rescue the
bem1d phenotype. So here when we talk about positive interaction we refer
to the 2nd case. I said that because from the mathematical definition is
not clear who rescue which genotype , and that is important in order to
predict beneficial mutations in different backgrounds. So what I add with
this is info about existing experimental data that complements the output
from the mathematical model. For now you can treat them bidirectional but
it is not always true and also that make sense because the effects of
deleting different genes could have different impact on the network that
can make this pair  not commutable .
